Gems/Atom/Feature/Common/Assets/ShaderLib/Atom/Features/Shadow/ESM.azsli

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+ 
+#pragma once
+ 
+#include <Atom/Features/Math/Filter.azsli> // BilinearFilter()
+
+ 
+float SampleESM(const Texture2DArray<float> shadowMap, const SamplerState samp, const float3 uv, const float zReceiver, const float overdark)
+{
+	const float mipmaplevel = 0;
+	const float occluder = shadowMap.SampleLevel(samp,uv, mipmaplevel).r;
+	const float lit = exp((occluder - zReceiver) * overdark);
+	return lit;
+}
+
+float PCFFallbackForESM(const Texture2DArray<float> shadowMap, const float3 uv, const float zReceiver, const float texMapSize, const float invTexMapSize, const float overdark)
+{
+    const float tl = clamp(SampleESM(shadowMap, PassSrg::PointSampler, uv + float3( 0.0, 0.0, 0.0 ) * invTexMapSize, zReceiver, overdark),0.0,1.0);
```

Comment:
Worth checking in the future if this can be done with a linear sampler and a single sample without lost quality

Reply:
Agreed. I played with it a bit and it did change the visual appearance slightly. I suspect multiple bilinear taps would be the best looking solution. At least it is all encapsulated into a single function now.